crates/owl-ms-language-server/src/lib.rs

```diff
@@ -714,6 +717,47 @@ impl LanguageServer for Backend {
         Ok(None)
     }
 
+    async fn symbol(
+        &self,
+        params: WorkspaceSymbolParams,
+    ) -> Result<Option<Vec<SymbolInformation>>> {
+        let query = params.query;
+        info!("symbol with query:{query}");
+        let workspaces = self.workspaces.lock().await;
+        let c = workspaces
+            .iter()
+            .flat_map(|w| {
+                w.document_map.iter()
+                // .map(|d| d.value().clone())
+                // .flat_map(|d| d.frame_infos.iter().map(|i| i.value())) //.iter().map(|i| i.value()))
+            })
+            .collect_vec();
+        let d = c.iter().flat_map(|a| a.frame_infos.iter()).collect_vec();
+
+        info!(
+            "All frame infos: {:#?}",
+            d.iter().map(|v| v.value()).collect_vec()
+        );
+
+        let symbols = d
+            .iter()
+            .filter(|i| i.iri.contains(query.as_str()))
+            .flat_map(|fi| {
+                fi.definitions.iter().map(|definition| SymbolInformation {
+                    name: fi.iri.clone(),
+                    kind: fi.frame_type.into(),
+                    tags: None,
+                    deprecated: None,
```

Comment:
The OWL standard supports deprecating entities.

>  An annotation with the `owl:deprecated` annotation property and the value equal to `"true"^^xsd:boolean` can be used to specify that an IRI is deprecated. 

Would this be difficult to add?

Reply:
Great idea